pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/WhenDirective.java

```diff
@@ -81,12 +84,29 @@ public String getDisplayName() {
         public String toGroovy(@Nonnull WhenDirective directive) {
             if (directive.conditional != null) {
                 UninstantiatedDescribable ud = UninstantiatedDescribable.from(directive.conditional);
-                DescribableModel model = ud.getModel();
+                DescribableModel<? extends DeclarativeStageConditional> model = ud.getModel();
                 if (model != null) {
+                    Map<String,Object> args = new HashMap<>();
+                    // Hack to null out non-required fields for UI prettiness.
+                    for (String argKey : ud.getArguments().keySet()) {
+                        DescribableParameter p = model.getParameter(argKey);
+                        if (p == null ||
+                                p.isRequired() ||
+                                !(String.class.equals(p.getErasedType())) ||
```

Comment:
I don't think I've seen the use of `String.class.equals` before, looks weird. `String` is final so I don't think you need anything complicated to check if something is a string or not.